pkg/logentry/stages/template_test.go

```diff
@@ -125,6 +125,39 @@ func TestTemplateStage_Process(t *testing.T) {
 				"missing":    "newval",
 			},
 		},
+		"template with multiple keys": {
+			TemplateConfig{
+				Source:   "message",
+				Template: "{{.Value}} in module {{.module}}",
+			},
+			map[string]interface{}{
+				"level":   "warn",
+				"app":     "loki",
+				"message": "warn for app loki",
+				"module":  "test",
+			},
+			map[string]interface{}{
+				"level":   "warn",
+				"app":     "loki",
+				"module":  "test",
+				"message": "warn for app loki in module test",
+			},
+		},
+		"template with multiple keys with missing source": {
+			TemplateConfig{
+				Source:   "missing",
+				Template: "{{ .level }} for app {{ .app | ToUpper }}",
```

Comment:
Add a test case where a template key  is missing from the template data just so we can see the expectations.